x-pack/plugins/security_solution/public/flyout/document_details/right/components/related_alerts_by_same_source_event.tsx

```diff
@@ -31,24 +45,61 @@ export const RelatedAlertsBySameSourceEvent: React.VFC<RelatedAlertsBySameSource
   originalEventId,
   scopeId,
 }) => {
+  const { eventId, indexName, isPreviewMode } = useDocumentDetailsContext();
+  const { openLeftPanel } = useExpandableFlyoutApi();
+
   const { loading, dataCount } = useFetchRelatedAlertsBySameSourceEvent({
     originalEventId,
     scopeId,
   });
-  const text = (
-    <FormattedMessage
-      id="xpack.securitySolution.flyout.right.insights.correlations.sourceAlertsLabel"
-      defaultMessage="{count, plural, one {alert} other {alerts}} related by source event"
-      values={{ count: dataCount }}
-    />
+
+  const onClick = useCallback(() => {
+    openLeftPanel({
```

Comment:
what do you think about abstracting all the left panel calls into a hook/helper function? I see quite a bit of redundancy like the params and keys. kind of like the cell actions
Just a thought, could do it in this PR or a separate refactor, wdyt?

Reply:
I'm not against abstracting, I just wonder to what extend do we push this?
- it seems that the `id` prop is always the same indeed.
- for the `path` prop I think it's different enough that it would be passed in to that hook/helper method
- for the `params` prop, we mostly use `eventId`, `scopeId` and `indexName`, and while those are mainly retrieved from the `useDocumentDetailsContext` hook, there are instances where they're passed down via props to the component. Would we have then as props to the hook/helper method and default those values internally using the `useDocumentDetailsContext` if they're undefined?